pkg/types/vsphere/validation/platform_test.go

```diff
@@ -38,6 +38,15 @@ func TestValidatePlatform(t *testing.T) {
 			}(),
 			expectedError: `^test-path\.vCenter: Required value: must specify the name of the vCenter$`,
 		},
+		{
+			name: "invalid vcenter name",
+			platform: func() *vsphere.Platform {
+				p := validPlatform()
+				p.VCenter = "http://forest.com:9090?param=value"
```

Comment:
this test case is failing because it includes params. it would be better to include the actual error message so that is clear, and also to include another test case that fails for transport

Reply:
I tried different regex strings. I had no luck with the test accepting them. The actual error message has full message which says there should not be params/scheme in vCenter. For the test I shortened it.